### PR TITLE
Add support to read rise time from SRF

### PR DESCRIPTION
**Pull request by lispandfound**
This PR adds SRF parsing support to extract the rise time. This is required for new source modelling plots, it's also just nice to have. It also catches an edge-case for the `.geometry` property to return a `shapely.LineString` instead of a `shapely.Polygon` when `dip = 90`.

### CODE REVIEW

**Comment by lispandfound** [comment]
Finally fixed all the geometry test cases, so this is ready for review.

**Review by joelridden — DISMISSED** [review]

**Comment by lispandfound** [comment]
Sorry @joelridden I just improved some slip tests to make them slightly faster.

File: tests/test_slip.py [hunk 1]
```diff
@@ -1,43 +1,42 @@
...
+    t0=st.floats(0.1, 100), offset=st.floats(1e-6, 100), total_slip=st.floats(1, 1000)
+)
+def test_box_car_slip(t0: float, offset: float, total_slip: float):
     """Check that the box car slip function is a well-behaved slip function.
 
     For the box car slip function we must have:
 
     1. Slip is always non-negative,
-    2. Slip is zero at t0 and t1,
     3. Slip function integrates to total slip over the interval [t0, t1].
```
Comment (AndrewRidden-Harper) [inline tests/test_slip.py:18]:
```suggestion
    2. Slip function integrates to total slip over the interval [t0, t1].
```

**Review by AndrewRidden-Harper — DISMISSED** [review]

**Review by AndrewRidden-Harper — APPROVED** [review]

**Review by joelridden — APPROVED** [review]
